File: ma_yisheng/test-go-cases/vulnerable.go
package main

import (
	"database/sql"
	"net/http"
	"os"
	"os/exec"
)

func commandInjectionCase(r *http.Request) {
	cmd := r.FormValue("cmd")
	exec.Command(cmd).Run()
}

func sqlInjectionCase(r *http.Request, db *sql.DB) {
	id := r.FormValue("id")
	query := "SELECT * FROM users WHERE id = " + id
	db.Query(query)
}

func pathTraversalCase(r *http.Request) {
	fileName := r.FormValue("file")
	os.Create(fileName)
}

func ssrfCase(r *http.Request) {
	url := r.FormValue("url")
	http.Get(url)
}

func main() {
	var r *http.Request
	var db *sql.DB

	commandInjectionCase(r)
	sqlInjectionCase(r, db)
	pathTraversalCase(r)
	ssrfCase(r)
}

File: ma_yisheng/test-js-cases/vulnerable.js
const child_process = require('child_process');
const fs = require('fs');
const axios = require('axios');

function commandInjectionCase(req, res) {
  const cmd = req.query.cmd;
  child_process.exec(cmd);
  res.send(cmd);
}

function sqlInjectionCase(req, connection) {
  const id = req.query.id;
  const sql = 'SELECT * FROM users WHERE id = ' + id;
  connection.query(sql);
}

function pathTraversalCase(req) {
  const fileName = req.query.file;
  fs.readFileSync(fileName);
}

function ssrfCase(req) {
  const url = req.query.url;
  axios.get(url);
}

module.exports = {
  commandInjectionCase,
  sqlInjectionCase,
  pathTraversalCase,
  ssrfCase,
};
